src/main/java/com/rackspace/jenkins_nodepool/NodePoolJob.java

```diff
@@ -68,8 +83,72 @@ public int hashCode() {
         return Long.hashCode(taskId);
     }
 
+    public boolean isDone() {
+        return getCurrentAttempt().isDone();
+    }
+
+    public boolean isSuccess() {
+        return getCurrentAttempt().isSuccess();
+    }
+
+    public String getDurationSeconds() {
+        try {
+            final long start = attempts.get(0).startTime;
+
+            long end;
+            if (isDone()) {
+                end = getCurrentAttempt().finishTime;
+            } else {
+                end = System.currentTimeMillis();
+            }
+
+            return Long.toString((end - start)/ 1000L);
+
+        } catch (IndexOutOfBoundsException e) {
+            // no attempts made yet
+            return null;
+        }
+
+    }
+
+    public NodePool getNodePool() {
+        try {
+            return getCurrentAttempt().request.nodePool;
+        } catch (IndexOutOfBoundsException e) {
+            return null;
+        }
+    }
+
+    /**
+     * Get the jenkins build number for the project, if it's known.
+     *
+     * Builds that are not yet "running" do not have a build number.
+     *
+     * @return build number of Jenkins project
+     */
+    public String getBuildNumber() {
+        if (buildNumber == null) {
+            return "";
+        } else {
+            return "#" + String.valueOf(buildNumber);
+        }
+    }
+
+    public Status getResult() {
+        return getCurrentAttempt().getResult();
+    }
+
+    public void setBuildNumber(int buildNumber) {
+        this.buildNumber = buildNumber;
+    }
+
+    public enum Status {
+        INPROGRESS,
+        SUCCESS,
+        FAILURE
+    }
```

Comment:
No PENDING state needed? Just wondering.

Reply:
The listener doesn't discover new Tasks until they are waiting to be scheduled on a Jenkins node.  From the perspective of the plugin it's already INPROGRESS at this point since it can begin actively provisioning nodes with NodePool.  I'm not sure how we'd include a PENDING state.

Reply:
👍